streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/BaseRepartitionNode.java

```diff
@@ -0,0 +1,97 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream.internals.graph;
+
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.streams.kstream.internals.KeyValueStoreMaterializer;
+import org.apache.kafka.streams.kstream.internals.MaterializedInternal;
+import org.apache.kafka.streams.processor.FailOnInvalidTimestamp;
+import org.apache.kafka.streams.processor.internals.InternalTopologyBuilder;
+
+public abstract class BaseRepartitionNode<K, V> extends StreamsGraphNode {
+
+    protected final Serde<K> keySerde;
+    protected final Serde<V> valueSerde;
+    protected final String sinkName;
+    protected final String sourceName;
+    protected final String repartitionTopic;
+    protected final ProcessorParameters processorParameters;
+    protected final MaterializedInternal materializedInternal;
+
+
+    BaseRepartitionNode(final String nodeName,
+                        final String sourceName,
+                        final ProcessorParameters processorParameters,
+                        final Serde<K> keySerde,
+                        final Serde<V> valueSerde,
+                        final String sinkName,
+                        final String repartitionTopic,
+                        final MaterializedInternal materializedInternal) {
+
+        super(nodeName, false);
+
+        this.keySerde = keySerde;
+        this.valueSerde = valueSerde;
+        this.sinkName = sinkName;
+        this.sourceName = sourceName;
+        this.repartitionTopic = repartitionTopic;
+        this.processorParameters = processorParameters;
+        this.materializedInternal = materializedInternal;
+
+    }
+
+    @Override
+    public void writeToTopology(InternalTopologyBuilder topologyBuilder) {
+        final Serializer<K> keySerializer = keySerde != null ? keySerde.serializer() : null;
+        final Deserializer<K> keyDeserializer = keySerde != null ? keySerde.deserializer() : null;
+
+
+        topologyBuilder.addInternalTopic(repartitionTopic);
+
+        topologyBuilder.addSink(sinkName,
+                                repartitionTopic,
+                                keySerializer,
+                                getValueSerializer(),
```

Comment:
There are 2 difference between the repartition operations the first is the `Serializer` and `Deserializer` required